dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultQueryItemLocator.java

```diff
@@ -211,8 +216,41 @@ else if ( type != null && type.equals( EventOutputType.ENROLLMENT ) )
 
     private ProgramStage getProgramStageOrFail( String dimension )
     {
-        BaseIdentifiableObject baseIdentifiableObject = getIdObjectOrFail( dimension );
-        return (baseIdentifiableObject instanceof ProgramStage ? (ProgramStage) baseIdentifiableObject : null);
+        BaseIdentifiableObject baseIdentifiableObject = getIdObjectOrFail( removeOffset( dimension ) );
+
+        return (baseIdentifiableObject instanceof ProgramStage
+            ? (ProgramStage) baseIdentifiableObject
+            : null);
+    }
+
+    private static String removeOffset( String dimension )
+    {
+        final Pattern pattern = Pattern.compile( INDEX_REGEX );
+
+        final Matcher matcher = pattern.matcher( dimension );
+
+        if ( matcher.find() )
+        {
+            return dimension.replace( matcher.group( 0 ), "" );
+        }
+
+        return dimension;
+    }
+
+    private static int getProgramStageOffset( String dimension )
```

Comment:
Do we have a unit test for this logic?

Reply:
it is tested now, thanks this I found the bug in the QueryItemLocator class

Reply:
it is tested now